service/controller/chartconfig/v7/resource/chart/create.go

```diff
@@ -6,16 +6,31 @@ import (
 	"fmt"
 
 	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/controller/context/resourcecanceledcontext"
 	"k8s.io/helm/pkg/helm"
 
-	"github.com/giantswarm/chart-operator/service/controller/chartconfig/v6/key"
+	"github.com/giantswarm/chart-operator/pkg/annotation"
+	"github.com/giantswarm/chart-operator/service/controller/chartconfig/v7/key"
 )
 
 func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange interface{}) error {
 	customObject, err := key.ToCustomObject(obj)
 	if err != nil {
 		return microerror.Mask(err)
 	}
+
+	reason, reasonOk := customObject.Labels[annotation.CordonReasonAnnotationName]
+	until, untilOk := customObject.Labels[annotation.CordonUntilAnnotationName]
+
+	if reasonOk && untilOk {
```

Comment:
```suggestion
	if key.IsCordoned(customResource) {
```

I'd prefer to use a key function for this and have some basic tests for it.